File: remote/src/main/java/com/kelsos/mbrc/utilities/MediaButtonReceiver.java
package com.kelsos.mbrc.utilities;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import com.google.inject.Inject;
import roboguice.RoboGuice;

public class MediaButtonReceiver extends BroadcastReceiver {
  @Inject private MediaIntentHandler handler;

  @Override public void onReceive(Context context, Intent intent) {
    if (handler == null) {
      RoboGuice.getInjector(context).injectMembers(this);
    }

    handler.handleMediaIntent(intent);
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/service/mediasession/MediaButtonReceiver.kt
package com.kelsos.mbrc.service.mediasession

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import org.koin.core.component.KoinComponent
import timber.log.Timber

// TODO: Check if this receiver is still needed. Media3's MediaSession handles media button
//  events through RemotePlayer. This may be redundant but is kept for backward compatibility.
class MediaButtonReceiver :
  BroadcastReceiver(),
  KoinComponent {
  override fun onReceive(context: Context, intent: Intent) {
    Timber.v("Incoming %s", intent)
    if (intent.action != Intent.ACTION_MEDIA_BUTTON) {
      return
    }
    getKoin().get<MediaIntentHandler>().handleMediaIntent(intent)
  }
}
